test/pg-cdc/pg-cdc.td

```diff
@@ -0,0 +1,399 @@
+
+! SELECT * FROM no_such_user;
+Source error: File IO: db error: FATAL: role "no_such_user" does not exist
+
+> CREATE MATERIALIZED SOURCE "no_such_password"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=no_such_password dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_password;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
+
+> CREATE MATERIALIZED SOURCE "no_such_dbname"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=no_such_dbname'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_dbname;
+Source error: File IO: db error: FATAL: database "no_such_dbname" does not exist
+
+> CREATE MATERIALIZED SOURCE "no_such_publication"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'no_such_publication' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_publication;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
+
+> CREATE MATERIALIZED SOURCE "no_such_namespace"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'no_such_namespace' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_namespace;
+Source error: File IO: table not found in the upstream catalog
+
+> CREATE MATERIALIZED SOURCE "no_such_table"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'no_such_table' (pk INTEGER, f2 TEXT);
+
+! SELECT * FROM no_such_table;
+Source error: File IO: table not found in the upstream catalog
+
+> CREATE MATERIALIZED SOURCE "no_such_column"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (no_such_column INTEGER, f2 TEXT);
+
+# ! SELECT * FROM no_such_column;
+# error expected here but success is returned
+
+> CREATE MATERIALIZED SOURCE "column_type_mismatch"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM column_type_mismatch;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
+
+> CREATE MATERIALIZED SOURCE "extra_column"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 INTEGER, f2 TEXT, f3 INTEGER);
+
+# ! SELECT f3 IS NULL FROM extra_column;
+# some-error-expected-here
```

Comment:
Having more columns in the definition then in the Postgres table causes a panic:

```2021-03-08T13:46:28.232121Z ERROR panic: index out of bounds: the len is 2 but the index is 2
thread: timely:work-0
location: src/expr/src/scalar/mod.rs:626:48
version: 0.7.2-dev (c91727b6897814ae0e3e1269ef910cde8c4ff186)
backtrace:
   0: materialized::handle_panic
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/materialized/src/bin/materialized/main.rs:618:21
   1: core::ops::function::Fn::call
             at usr/local/lib/rustlib/src/rust/library/core/src/ops/function.rs:70:5
   2: std::panicking::rust_panic_with_hook
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:597:17
   3: std::panicking::begin_panic_handler::{{closure}}
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:499:13
   4: std::sys_common::backtrace::__rust_end_short_backtrace
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys_common/backtrace.rs:141:18
   5: rust_begin_unwind
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/panicking.rs:495:5
   6: core::panicking::panic_fmt
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/panicking.rs:92:14
   7: core::panicking::panic_bounds_check
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/core/src/panicking.rs:69:5
   8: expr::scalar::MirScalarExpr::eval
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/expr/src/scalar/mod.rs:626:48
   9: dataflow::render::arrange_by::<impl dataflow::render::context::Context<G,expr::relation::MirRelationExpr,repr::row::Row,T>>::render_arrangeby::{{closure}}::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/render/arrange_by.rs:48:64
      core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
             at usr/local/lib/rustlib/src/rust/library/core/src/ops/function.rs:280:13
      core::option::Option<T>::map
             at usr/local/lib/rustlib/src/rust/library/core/src/option.rs:453:29
      <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::next
             at usr/local/lib/rustlib/src/rust/library/core/src/iter/adapters/mod.rs:924:9
      repr::row::RowPacker::try_extend
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/repr/src/row.rs:833:22
      repr::row::Row::try_pack
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/repr/src/row.rs:578:9
  10: dataflow::render::arrange_by::<impl dataflow::render::context::Context<G,expr::relation::MirRelationExpr,repr::row::Row,T>>::render_arrangeby::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/render/arrange_by.rs:48:29
      <differential_dataflow::collection::Collection<G,D1,R> as dataflow::operator::CollectionExt<G,D1,R>>::map_fallible::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/operator.rs:334:87
      <timely::dataflow::stream::Stream<G,D1> as dataflow::operator::StreamExt<G,D1>>::map_fallible::{{closure}}::{{closure}}::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/operator.rs:229:31
      timely::dataflow::operators::generic::handles::InputHandle<T,D,P>::for_each
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/dataflow/operators/generic/handles.rs:92:13
      <timely::dataflow::stream::Stream<G,D1> as dataflow::operator::StreamExt<G,D1>>::map_fallible::{{closure}}::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/operator.rs:224:17
  11: <alloc::boxed::Box<F,A> as core::ops::function::FnMut<Args>>::call_mut
             at usr/local/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1314:9
      <timely::dataflow::stream::Stream<G,D1> as dataflow::operator::StreamExt<G,D1>>::unary_fallible::{{closure}}::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/operator.rs:208:17
      timely::dataflow::operators::generic::builder_rc::OperatorBuilder<G>::build::{{closure}}::{{closure}}
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/dataflow/operators/generic/builder_rc.rs:123:31
      timely::dataflow::operators::generic::builder_rc::OperatorBuilder<G>::build_reschedule::{{closure}}
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/dataflow/operators/generic/builder_rc.rs:163:26
      <timely::dataflow::operators::generic::builder_raw::OperatorCore<T,L> as timely::scheduling::Schedule>::schedule
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/dataflow/operators/generic/builder_raw.rs:203:9
  12: timely::progress::subgraph::PerOperatorState<T>::schedule
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/progress/subgraph.rs:656:30
  13: timely::progress::subgraph::Subgraph<TOuter,TInner>::activate_child
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/progress/subgraph.rs:338:26
      <timely::progress::subgraph::Subgraph<TOuter,TInner> as timely::scheduling::Schedule>::schedule
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/progress/subgraph.rs:304:17
  14: timely::progress::subgraph::PerOperatorState<T>::schedule
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/progress/subgraph.rs:656:30
  15: timely::progress::subgraph::Subgraph<TOuter,TInner>::activate_child
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/progress/subgraph.rs:338:26
      <timely::progress::subgraph::Subgraph<TOuter,TInner> as timely::scheduling::Schedule>::schedule
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/progress/subgraph.rs:304:17
  16: timely::worker::Wrapper::step::{{closure}}
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/worker.rs:727:57
      core::option::Option<T>::map
             at usr/local/lib/rustlib/src/rust/library/core/src/option.rs:453:29
      timely::worker::Wrapper::step
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/worker.rs:727:26
  17: timely::worker::Worker<A>::step_or_park
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/worker.rs:375:38
  18: dataflow::server::Worker<A>::run
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/server.rs:430:13
  19: dataflow::server::serve::{{closure}}
             at var/lib/buildkite-agent/builds/buildkite-builders-i-03dc3cbd65955f78f-1/materialize/tests/src/dataflow/src/server.rs:206:13
      timely::execute::execute::{{closure}}
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/timely/src/execute.rs:284:22
      timely_communication::initialize::initialize_from::{{closure}}
             at cargo/git/checkouts/timely-dataflow-4c0cc365061cd263/84b1116/communication/src/initialize.rs:285:33
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at usr/local/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:125:18
  20: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at usr/local/lib/rustlib/src/rust/library/std/src/thread/mod.rs:474:17
      <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at usr/local/lib/rustlib/src/rust/library/std/src/panic.rs:322:9
      std::panicking::try::do_call
             at usr/local/lib/rustlib/src/rust/library/std/src/panicking.rs:381:40
      std::panicking::try
             at usr/local/lib/rustlib/src/rust/library/std/src/panicking.rs:345:19
      std::panic::catch_unwind
             at usr/local/lib/rustlib/src/rust/library/std/src/panic.rs:396:14
      std::thread::Builder::spawn_unchecked::{{closure}}
             at usr/local/lib/rustlib/src/rust/library/std/src/thread/mod.rs:473:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at usr/local/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
  21: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/alloc/src/boxed.rs:1307:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/alloc/src/boxed.rs:1307:9
      std::sys::unix::thread::Thread::new::thread_start
             at rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca/library/std/src/sys/unix/thread.rs:71:17
  22: start_thread
  23: __clone
```

Reply:
It seems that the ```CREATE SOURCE``` borks the entire database, future selects on unrelated tables panic as above.